jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ClientConnector.java

```diff
@@ -629,6 +634,7 @@ protected void connectionFailed(SelectableChannel channel, Throwable failure, Ob
         {
             @SuppressWarnings("unchecked")
             Map<String, Object> context = (Map<String, Object>)attachment;
+            ClientConnector.this.getBeans(ClientConnector.ConnectListener.class).forEach(listener -> listener.onConnectFailure((SocketChannel)channel, failure));
```

Comment:
Shouldn't this be moved inside `connectFailed()`?

Reply:
It leads to changing of signature of `connectFailed()` method (3 parameters instead of 2 ones)

Reply:
@arsenalzp that's ok to pass the channel as first parameter.